runtime/opt/taupage/init.d/10-prepare-disks.py

```diff
@@ -30,6 +31,29 @@ def zone():
     return boto.utils.get_instance_metadata()['placement']['availability-zone']
 
 
+def select_keys(d, keys):
+    """
+    >>> select_keys({'foo': 1, 'bar': 2, 'baz': 'def'}, ['foo', 'baz'])
+    {'foo': 1, 'baz': 'def'}
+
+    >>> select_keys({'foo': 1, 'bar': 2}, ['foo', 'baz'])
+    {'foo': 1}
+    """
+    return {
+        k: d[k]
+        for k in keys
+        if k in d
+    }
+
+
+def allowed_volume_name_subs():
+    """Helper to return the allowed volume name substitutions from the instance metadata"""
+    return select_keys(
+        boto.utils.get_instance_metadata(),
```

Comment:
IMDS has hiccups sometimes. Does this retry?

Reply:
Although I guess it's used in other parts of the script anyway, so it won't be worse.

Reply:
I've also noticed that `find_volume` isn't annotated with `@retry`.  Should we add it to this PR or make a separate one just for that?

Reply:
Let's do it separately.